utils/splunk/splunk_helper.py

```diff
@@ -102,6 +102,30 @@ def dispatch(self, saved_search, splunk_user, splunk_app, splunk_ignore_saved_se
         response_body = self._do_post(dispatch_path, parameters, saved_search.request_timeout_seconds, splunk_ignore_saved_search_errors).json()
         return response_body.get("sid")
 
+    def finalize_sid(self, search_id, saved_search):
+        """
+        :param search_id: The saved search id to finish
+        :param saved_search: The saved search to finish
+        """
+        finish_path = '/services/search/jobs/%s/control' % (search_id)
+        payload = "action=finalize"
+        try:
+            res = self._do_post(finish_path, payload, saved_search.request_timeout_seconds, splunk_ignore_saved_search_errors=False)
+            if res.status_code == 200:
+                self.log.info("Saved Search ID %s finished successfully." % search_id)
+            if res.status_code == 404:
+                self.log.info("Saved Search ID %s is not found." % search_id)
+        except HTTPError:
```

Comment:
When is this error thrown? Would be good to have a comment here what can cause this